mage_ai/frontend/pages/overview/index.tsx

```diff
@@ -130,16 +125,60 @@ function OverviewPage() {
     group_by_pipeline_type: 1,
     start_time: startDateString,
   }), [startDateString]);
-  const {
-    data: dataMonitor,
-    isValidating: isValidatingMonitorStats,
-    mutate: fetchMonitorStats,
-  } = api.monitor_stats.detail(
-    MonitorStatsEnum.PIPELINE_RUN_COUNT,
-    monitorStatsQueryParams,
-    { ...SHARED_FETCH_OPTIONS },
+
+  const [monitorStats, setMonitorStats] = useState();
+  const [fetchMonitorStats, { isLoading: isValidatingMonitorStats }] = useMutation(
+    () => api.monitor_stats?.detailAsync(
+      MonitorStatsEnum.PIPELINE_RUN_COUNT,
+      monitorStatsQueryParams,
+      {
+        signal: abortRef?.current?.signal,
+      },
+    ),
+    {
+      onSuccess: (response: any) => {
+        return onSuccess(
+          response,
+          {
+            callback: ({
+              monitor_stat: {
+                stats,
+              },
+            }) => {
+              setMonitorStats(stats);
+            },
+          },
+        );
+      },
+    },
   );
 
+  const setSelectedTab = useCallback((prev: TabType | ((tab: TabType) => TabType)) => {
+    if (abortRef?.current !== null) {
+      abortRef?.current?.abort();
+    }
+    abortRef.current = new AbortController();
+
+    setSelectedTabState((current: TabType) => {
+      const tab = typeof prev === 'function' ? prev(current) : prev;
+
+      if (current?.uuid !== tab?.uuid) {
+        goToWithQuery({ [TAB_URL_PARAM]: tab?.uuid }, { replaceParams: true });
+
+        fetchMonitorStats();
+      }
+
+      return tab;
+    });
+  }, []);
+
+  useEffect(() => {
```

Comment:
@dy46 @johnson-mage let’s not make API calls in `useEffect` anymore; it’ll make multiple unnecessary API calls.

Reply:
got it